kernel/common/enumerable.rb

```diff
@@ -314,6 +314,28 @@ def grep(pattern)
     ary
   end
 
+  def grep_v(pattern)
+    ary = []
+
+    if block_given?
```

Comment:
The overhead of `block_given?` isn't that dramatic so I think this can just be changed to something like:

```ruby
if block_given?
  ary << yield element
else
  ary << element
end
```